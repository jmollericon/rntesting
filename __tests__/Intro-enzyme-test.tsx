import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import Intro from '../src/components/Intro';

configure({ adapter: new Adapter() });

describe('Testing Intro component - enzyme', () => {
  it('renders as expected', () => {
    const wrapper = shallow(<Intro count={2} />);
    expect(wrapper).toMatchSnapshot();
    wrapper.setProps({ count: 1 });
    expect(wrapper).toMatchSnapshot();
  });
});
