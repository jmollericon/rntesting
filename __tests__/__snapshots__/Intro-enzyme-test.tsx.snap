// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Intro component - enzyme renders as expected 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "margin": 10,
        "textAlign": "center",
      }
    }
  >
    Welcome to React Native!
  </Text>
  <Text
    style={
      Object {
        "color": "#333333",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }
  >
    This is a React Native snapshot test. 
    2
  </Text>
</View>
`;

exports[`Testing Intro component - enzyme renders as expected 2`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "margin": 10,
        "textAlign": "center",
      }
    }
  >
    Welcome to React Native!
  </Text>
  <Text
    style={
      Object {
        "color": "#333333",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }
  >
    This is a React Native snapshot test. 
    1
  </Text>
</View>
`;
